src/huggingface_hub/utils/_git_credential.py

```diff
@@ -0,0 +1,202 @@
+# coding=utf-8
+# Copyright 2022-present, the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains utilities to manage Git credentials."""
+import subprocess
+from typing import List, Optional, Tuple, Union
+
+from ..constants import ENDPOINT
+from ._deprecation import _deprecate_method
+from ._subprocess import run_interactive_subprocess, run_subprocess
+
+
+def list_credential_helpers(directory: Optional[str] = None) -> List[str]:
+    """Return the list of git credential helpers configured.
+
+    See https://git-scm.com/docs/gitcredentials.
+
+    Credentials are saved in all configured helpers (store, cache, macos keychain,...).
+    Calls "`git credential approve`" internally. See https://git-scm.com/docs/git-credential.
+
+    Args:
+        directory (`str`, *optional*):
+            The directory in which to check the configured helpers.
+    """
+    try:
+        output = run_subprocess("git config --list", directory).stdout
+        return [
+            line.split("=")[-1]
+            for line in output.split("\n")
+            if "credential.helper" in line
+        ]
```

Comment:
For this, just a small Q: you're not taking into account the site-specific git credential helper with this approach: `git config --list` returns the default credential helper as you have seen, but if a user has set a custom helper for hf.co, it might return:
```
credential.https://huggingface.co.helper=
credential.https://huggingface.co.helper=store
```
which wouldn't be handled by your `if "credential.helper" in line`

for example. I don't think that's necessarily important, just putting it down in case you weren't aware. Thanks!

Reply:
Wow, definitely didn't know about that ! I just took over the existing code without changing the logic.

I don't think I'll change the implementation to handle this but just in case the question arise in the future, I have added a warning comment that links to this issue for context.